modal/_runtime/user_code_imports.py

```diff
@@ -334,8 +334,22 @@ def import_class_service(
         # that transfers metadata from the service function to the class. TODO: cleanup!
         _cls._hydrate(class_id, _client, api_pb2.ClassHandleMetadata())
 
+    # UPDATE TO USE with_http_config from ImportedClass in USER_CODE_IMPORTs.py
+    # Instead of extracting from experimental_options, extract from protobuf:
+    from modal._partial_function import _HTTPConfig
+
+    http_config: Optional[_HTTPConfig] = None
+    if function_def.http_config:
+        http_config = _HTTPConfig(
+            port=function_def.http_config.port,
+            proxy_region=function_def.http_config.proxy_region,  # type: ignore
+            startup_timeout=function_def.http_config.startup_timeout or None,
+            exit_grace_period=function_def.http_config.exit_grace_period or None,
+        )
+        _cls._options.http_config = http_config  # type: ignore
```

Comment:
Why are we writing to the `_cls._options` here? That's normally used client side for `modal.Cls.with_options`; I don't think we have any precedent for using it to hold state inside the container runtime.